### PR TITLE
feat/introduce cache related metrics for prometheus

### PR DESCRIPTION
**Pull request by firatNEAR**
Part 1 of cache metrics for monitoring.
Related issue https://github.com/near/nearcore/issues/7375

### CODE REVIEW

File: core/store/src/trie/trie_storage.rs [hunk 1]
```diff
@@ -43,10 +46,20 @@ impl TrieCache {
                         guard.put(hash, value.into());
                     }
```
Comment (Longarithm) [inline core/store/src/trie/trie_storage.rs:47]:
```suggestion
                    } else {
                        metrics::SHARD_CACHE_TOO_LARGE.with_label_values(&labels).inc();
                    }
```
We need to inc this metric here as well - this is my bug from the first impl

File: core/store/src/trie/trie_storage.rs [hunk 2]
```diff
@@ -34,7 +34,10 @@ impl TrieCache {
         self.0.lock().expect(POISONED_LOCK_ERR).clear()
     }
 
-    pub fn update_cache(&self, ops: Vec<(CryptoHash, Option<&Vec<u8>>)>) {
+    pub fn update_cache(&self, ops: Vec<(CryptoHash, Option<&Vec<u8>>)>, shard_uid: ShardUId) {
```
Comment (Longarithm) [inline core/store/src/trie/trie_storage.rs:37]: Now I think that `TrieCache` should hold `shard_uid`, so we could avoid passing it here. Let's fix it in a separate PR

File: core/store/src/metrics.rs [hunk 3]
```diff
@@ -10,3 +13,62 @@ pub(crate) static DATABASE_OP_LATENCY_HIST: Lazy<HistogramVec> = Lazy::new(|| {
...
+        "Shard cache misses",
+        &["shard_id", "is_view"],
+    )
+    .unwrap()
+});
+
+pub static SHARD_CACHE_TOO_LARGE: Lazy<IntCounterVec> = Lazy::new(|| {
+    try_create_int_counter_vec(
+        "near_shard_cache_too_large",
+        "Shard cache too large",
```
Comment (Longarithm) [inline core/store/src/metrics.rs:56]:
```suggestion
        "Number of too large values to be inserted into shard cache",
```

File: core/store/src/trie/trie_storage.rs [hunk 4]
```diff
@@ -231,21 +252,36 @@ impl TrieCachingStorage {
...
+        let is_view_str = format!("{}", self.is_view as u8);
+        let labels: [&str; 2] = [&shard_id_str, &is_view_str];
+        {
+            metrics::CHUNK_CACHE_SIZE
+                .with_label_values(&labels)
+                .set(self.chunk_cache.borrow().len() as i64);
+            metrics::SHARD_CACHE_SIZE
+                .with_label_values(&labels)
+                .set(self.shard_cache.0.lock().expect(POISONED_LOCK_ERR).len() as i64);
+        }
```
Comment (Longarithm) [inline core/store/src/trie/trie_storage.rs:265]: Let's separate it to `fn update_cache_size_metrics()` for `TrieCachingStorage` and drop extra curly braces

**Review by Longarithm — APPROVED** [review]

**Comment by Longarithm** [comment]
We decided to merge it with https://github.com/near/nearcore/pull/7429, after finishing it we just close this one

File: core/store/src/trie/trie_storage.rs (hunk 4 above)
Comment (Longarithm) [reply core/store/src/trie/trie_storage.rs]: Sorry, I forgot that we use `labels` in this method below. Then the original code makes more sense

**Comment by matklad** [comment]
https://github.com/near/nearcore/pull/742 is merged, should we close this?

**Pull request closed without merging**